src/vehicle.cpp

```diff
@@ -64,6 +64,21 @@ const std::array<fuel_type, 7> &get_fuel_types()
     return fuel_types;
 }
 
+const std::unordered_map<itype_id, fuel_type> &get_fuel_types_table()
```

Comment:
This duplicates the content of `get_fuel_types()`, which is always a bad thing because those two containers *will* get out of sync.

Considering that the fuel types array is quite small, and you don't use the map in performance critical code, I suggest iteration over the the array instead. Something like this:
```C++
nc_color get_fuel_color( const itype_id &fuel_type ) {
    for( auto &f : get_fuel_types() ) {
        if( f.type == fuel_type ) {
            return f.color;
        }
    }
    return item::find_type( fuel_type )->color;
}
```
This encapsulates the whole logic nicely.